test/benchmarks/benchmark_adler32.cc

```diff
@@ -0,0 +1,125 @@
+/* benchmark_adler32.cc -- benchmark adler32 variants
+ * Copyright (C) 2020 Nathan Moinvaziri
+ * For conditions of distribution and use, see copyright notice in zlib.h
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <assert.h>
+
+#include <benchmark/benchmark.h>
+
+extern "C" {
+#  include "zbuild.h"
+#  include "zutil.h"
+#  include "cpu_features.h"
+}
+
+#define MAX_RANDOM_INTS (1024 * 1024)
+
+typedef uint32_t (*adler32_func)(uint32_t adler, const unsigned char *buf, size_t len);
+
+class adler32: public benchmark::Fixture {
+private:
+    uint32_t *random_ints;
+public:
+    void SetUp(const ::benchmark::State& state) {
+        int32_t random_ints_size = MAX_RANDOM_INTS * sizeof(uint32_t);
+
+        random_ints = (uint32_t *)malloc(random_ints_size);
+        assert(random_ints != NULL);
+
+        for (int32_t i = 0; i < MAX_RANDOM_INTS; i++) {
+            random_ints[i] = rand();
+        }
+    }
+
+    void Bench(benchmark::State& state, adler32_func adler32) {
+        uint32_t hash = 0;
+
+        for (auto _ : state) {
+            hash = adler32(hash, (const unsigned char *)random_ints, state.range(0) * sizeof(uint32_t));
+        }
+
+        benchmark::DoNotOptimize(hash);
+    }
+
+    void TearDown(const ::benchmark::State& state) {
+        free(random_ints);
+    }
+};
+
+
+BENCHMARK_DEFINE_F(adler32, c)(benchmark::State& state) {
+    Bench(state, adler32_c);
+}
+BENCHMARK_REGISTER_F(adler32, c)->Range(1, MAX_RANDOM_INTS);
```

Comment:
Looking good so far, but we should probably change this to reflect number of bytes rather than integer.  As in, remove the *sizeof(uint32_t) within the Bench function and modify range to be 1 to sizeof(uint32_t) * MAX_RANDOM_INTS.

Reply:
Good call. Done.